healthcareai/deploy_supervised_model.py

```diff
@@ -23,150 +25,152 @@ class DeploySupervisedModel(object):
     """
 
     def __init__(self,
-                 modeltype,
-                 df,
-                 graincol,
-                 windowcol,
-                 predictedcol,
+                 model_type,
+                 dataframe,
+                 grain_column,
+                 window_column,
+                 predicted_column,
                  impute,
                  debug=False):
-
-
-        self.modeltype = modeltype
-        self.df = df
-        self.graincol = graincol
+        self.modeltype = model_type
+        if model_type == 'classification':
+            self.predicted_column_name = 'PredictedProbNBR'
+        else:
+            self.predicted_column_name = 'PredictedValueNBR'
+        self.df = dataframe
+        self.grain_column = grain_column
         self.impute = impute
-        self.y_pred = None # Only used for unit test
+        self.y_pred = None  # Only used for unit test
 
         if debug:
             print('\nTypes of original dataframe:')
-            print(df.dtypes)
+            print(self.df.dtypes)
             print('\nShape and top 5 rows of original dataframe:')
-            print(df.shape)
-            print(df.head())
+            print(self.df.shape)
+            print(self.df.head())
 
         self.df = filters.remove_DTS_postfix_columns(self.df)
 
         if debug:
             print('\nDataframe after removing DTS columns:')
-            print(df.head())
+            print(self.df.head())
             print('\nNow either doing imputation or dropping rows with NULLs')
             print('\nTotal df shape before impute/remove')
-            print(df.shape)
+            print(self.df.shape)
 
         pd.options.mode.chained_assignment = None  # default='warn'
 
 
         # TODO: put try/catch here when type = class and pred col is numer
-        df.replace('NULL', np.nan, inplace=True)
+        self.df.replace('NULL', np.nan, inplace=True)
 
         if debug:
             print('\nTotal df after replacing NULL with nan')
-            print(df.shape)
-            print(df.head())
+            print(self.df.shape)
+            print(self.df.head())
 
         if self.impute:
 
-            df = DataFrameImputer().fit_transform(df)
+            self.df = DataFrameImputer().fit_transform(self.df)
 
             if debug:
                 print('\nTotal df after doing imputation:')
-                print(df.shape)
-                print(df.head())
+                print(self.df.shape)
+                print(self.df.head())
 
         else:
             # TODO switch similar statements to work inplace
             # If not impute, only remove rows in train that contain NULLS
 
             if debug:
                 print('\nTrain portion before impute/remove')
-                print(df.ix[df[windowcol] == 'N'].shape)
+                print(self.df.ix[self.df[window_column] == 'N'].shape)
```

Comment:
Consider saving `window_column` as an attribute and then referencing it as `self.window_column` throughout the code.  This will make it easier to refactor this method later - you'll be able to lift code out wholesale rather than moving and _also_ renaming variables.

Also, if you consider the variable to be 'protected' (or not useful to any external classes), consider naming the variable with a leading underscore.  So, it would be `self._window_column`.  This is a pretty typical python naming convention.